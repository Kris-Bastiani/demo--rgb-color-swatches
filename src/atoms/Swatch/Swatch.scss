.swatch {
	display: inline-block;
	margin: 1rem;
	border: 0;
	padding: 0.5rem;
	width: 12rem;
	font: inherit;
	text-align: center;
	background: none;
	box-shadow: 0 0 8px rgba(1, 2, 3, 0.2),
		0 0 2px rgba(1, 2, 3, 0.2);

	&__color {
		padding: 0 0 100%;
		width: 100%;
		height: 0;
	}

	&:hover,
	&:focus { outline: 2px solid rgb(1, 2, 3); }
}
