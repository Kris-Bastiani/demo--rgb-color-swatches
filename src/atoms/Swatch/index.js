import React from 'react';

import './Swatch.scss';

const Swatch = ({ red, green, blue }) => {
	const rgb = `rgb(${red}, ${green}, ${blue})`;

	return (
		<button className='swatch'>
			<div className='swatch__color' style={{ backgroundColor: rgb }}></div>
			<p className='swatch__label'>{rgb}</p>
		</button>
	);
};

export default Swatch;
