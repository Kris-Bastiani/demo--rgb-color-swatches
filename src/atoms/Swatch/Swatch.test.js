import React from 'react';
import { render } from '@testing-library/react';

import Swatch from '.';

describe('Component: Swatch', () => {
	it('should match snapshot', () => {
		const { container } = render(<Swatch red={0} green={0} blue={0} setSelected={() => {}} />);
		expect(container).toMatchSnapshot();
	});
});
