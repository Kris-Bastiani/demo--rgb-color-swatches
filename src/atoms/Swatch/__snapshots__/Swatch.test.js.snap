// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component: Button should match snapshot 1`] = `
<div>
  <button
    class="swatch"
  >
    <div
      class="swatch__color"
      style="background-color: rgb(0, 0, 0);"
    />
    <p
      class="swatch__label"
    >
      rgb(0, 0, 0)
    </p>
  </button>
</div>
`;
